app/api/questions.js

```diff
@@ -0,0 +1,132 @@
+const router = require('express').Router({ mergeParams: true })
+const db = require('../models/index')
+const { Op, ValidationError } = require("sequelize");
+const questionService = require('../services/question_service')
+const { requireAuthentication } = require('../../lib/auth')
+const string_helpers = require('../../lib/string_helpers')
+
+// GET /courses/course_id/questions?search=""&page=""&perPage=""
+// gets all the questions for a given course
+router.get('/', requireAuthentication, async function (req, res, next) {
+    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
+    const courseId = parseInt(req.params['course_id'])
+    // check to make sure the user is a teacher for the specified course
+    const enrollmentTeacher = await db.Enrollment.findOne({
+        where: { 
+            userId: user.id,
+            courseId: courseId,
+            role: 'teacher'
+        }
+    })
+
+    if (enrollmentTeacher) {
+        // we want search to have a value even if it is left out of the url
+        const search = (req.query.search != null) ? req.query.search : ""
+        // page and perPage are required query string parameters
+        const page = (req.query.page != null) ? parseInt(req.query.page) : null
+        const perPage = (req.query.perPage != null) ? parseInt(req.query.perPage) : null
+        if (page != null && perPage != null) {
+            try {
+                // get the questions needed
+                const questions = await db.Question.findAll({
+                    limit: perPage,
+                    offset: page * perPage,
+                    where: {
+                        stem: {
+                            [Op.like]: '%' + search + '%'
+                        },
+                        courseId: courseId
+                    }
+                })
+                // get the total questions for the course so that page number calculations can be made
+                const totalQuestions = await db.Question.findAll({
+                    where: {
+                        stem: {
+                            [Op.like]: '%' + search + '%'
+                        },
+                        courseId: courseId
+                    }
+                })
+                if (page * perPage < totalQuestions.length) {
```

Comment:
Don't you need to do a round up or something here? What if there are 49 total questions, you're requesting page number 2, and you're requesting 25 per page? It would error but that doesn't make sense.

Reply:
I thought I left a comment on this in the code but page numbers are indexed like an array (as in they start at 0). So if you were on page "2" page would equal 1. And if you did try to have page = 2 then there would be an out of bounds error